File: js-pr/56/javascript/code-quality-error-handling/file_processor_26.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
function readFile(filename) {
    const fs = require('fs');
    try {
        return fs.readFileSync(filename, 'utf8');
    } catch (error) {
        console.error(`Error reading file ${filename}:`, error.message);
        return null;
    }
}
// {/fact}

File: python-pr/52/python/code-quality-error-handling/file_processor_26.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def read_file(filename):
    try:
        with open(filename, 'r') as f:
            return f.read()
    except IOError as e:
        print(f"Error reading file {filename}: {e}")
        return None
# {/fact}